projects/rero/ng-core/src/lib/record/editor/editor.component.ts

```diff
@@ -495,6 +495,31 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
           };
           delete formOptions.validation.validators.valueAlreadyExists;
         }
+        // asyncValidators: valueKeysInObject
+        //  This validator is similar to uniqueValidator but only check on some specific fields of array items.
+        if (formOptions.validation.validators.uniqueValueKeysInObject) {
+          field.validators = {
+            uniqueValueKeysInObject: {
+              expression: (control: FormControl) => {
+                // if value isn't an array or array contains less than 2 elements, no need to check
+                if (control instanceof Array || control.value.length < 2) {
```

Comment:
The comment (`value isn't an array`) and the condition (`control instanceof Array`) seem to be contradictory.

Reply:
You're absolutely right ! Corrected